### ISSUE TITLE
The documented Oauth2 based authentication is wrong

### ISSUE THREAD

**Issue by giohappy** [geosolutions-it/nexus-geonode#580]
After spending some time reviewing the details of the Oauth2 based setup I've found that it's based on totally wrong assumptions.

If an Oauth2 client application is configured the issued tokens are for that application. However, Geoserver expects to receive its own access tokens to authenticate user requests against GeoNode. If a QGIS Client access token is used to make direct requests to Geoserver they won't be authenticated, since GeoNode will not match the token with any Geoserver Client token.

You see in the picture below that QGIS is using its own client's token to issue the request, From Django admin, we can see that that token is associated with QGIS Client. In the end, the authorization between Geoserver and GeoNode will fail.

The way tokens for Geoserver requests are managed would require setting up the Geoserver Oauth2 client inside QGIS, but it's not feasible since it would open up the internal Geoserver client secrets to the world.

I know that the way GeoNode uses and issues Geoserver access tokens is quite non-standard. It's the way it is for several reasons that date far away in the past.
Anyway, it also makes sense that, according to standard Oauth2 flows, QGIS cannot receive from GeoNode a Geoserver token  being two distinct services.

Long story short, the only authentication that should be documented is the Basic Auth. 

![image](https://user-images.githubusercontent.com/571129/154701748-57b60fa0-58e6-4a56-95ab-09541015999f.png)

![image](https://user-images.githubusercontent.com/571129/154701786-555c7bfc-4248-49fc-af5e-b5083b20f8e9.png)

**Comment by ricardogsilva** [comment]
I need to contest this, since it does not match with my dev experience while working on the plugin.

My practical experience tels me that, in its current form, OAuth2 authorization in the QGIS GeoNode plugin works OK. On my local dev environment I am able to use the OAuth2 authorization code flow in QGIS and gain access to private GeoNode resources.

Additionally, from a documentation side, [section 7 of the OAuth2 spec](https://datatracker.ietf.org/doc/html/rfc6749#section-7) states the following:

> The client accesses protected resources by presenting the access token to the resource server.  The resource server MUST validate the access token and ensure that it has not expired and that its scope covers the requested resource.  The methods used by the resource server to validate the access token (as well as any error responses) are beyond the scope of this specification but generally involve an interaction or coordination between the resource server and the authorization server.

As stated in the spec, the way used by the resource server (which in our case is GeoServer) to validate the access token presented by the client (which in our case is QGIS) is left as an implementation detail. However, this validation is stated to generally imply further contact between the resource server and the authorization server (which in our case is GeoNode).

The way that it seems to work is that GeoServer, upon receiving a request with a certain access token, contacts GeoNode in order to verify if the access token is valid. GeoNode then proceeds to validate that the token is indeed valid and lets GeoServer know that authorization is OK and access to the resources (in this case, some WMS/WFS/WCS request) should be granted. 

If this is indeed how it is working (and this matches my intuition, but I've not checked the source code) then it seems like a normal OAuth2 flow to me.

Now, having said that, I do agree that setting up OAuth2 as an authorization method is much more complicated than Basic Auth (which is already working) and does not seem to provide much value. As such, I'm :+1: on de-emphasizing the section on OAuth2 auth even more in the plugin docs. I think I'll put it in a collapsible side note

**Comment by giohappy** [comment]
@ricardogsilva I was wrong. I was sure that GeoNode only validated the tokens issued for Geoserver, but I'm reviewing the validation and, apparently, it validates any token from any registered Oauth2 client application.

By default GeoNode generates a token for Geoserver client application by default when a user logs in. This was confusing me, since I though that for this reason only tokens issued for Geoserver were accepted.

**Comment by ricardogsilva** [comment]
@giohappy FYI I updated the plugin docs and have de-emphasized OAuth2 as an authorization method even more. The instructions are now tucked inside a collapsible section (which is collapsed by default). I think that makes it pretty clear what the recommended setup is:

https://kartoza.github.io/qgis_geonode/user-guide/#configure-authentication
